modules/azure-mi/main.tf

```
@@ -30,6 +31,29 @@ resource "azurerm_user_assigned_identity" "this" {
   tags                = local.tags
 }
 
+## https://registry.terraform.io/providers/hashicorp/azurerm/4.19.0/docs/resources/role_definition
+resource "azurerm_role_definition" "custom_role" {
```

```suggestion
resource "azurerm_role_definition" "this" {
```

See https://www.terraform-best-practices.com/naming